authenticate/handlers.go

```diff
@@ -137,15 +138,21 @@ func (a *Authenticate) SignOut(w http.ResponseWriter, r *http.Request) {
 }
 
 // redirectToIdentityProvider starts the authenticate process by redirecting the
-// user to their respective identity provider.
-//
+// user to their respective identity provider. This function also builds the
+// 'state' parameter which is encrypted and includes authenticating data
+// for validation.
+// 'state' is : nonce|timestamp|redirect_url|encrypt(redirect_url)+mac(nonce,ts))
```

Comment:
Is this accurate?  I'm having trouble finding all of those fields being inserted.

Reply:
See : https://github.com/pomerium/pomerium/pull/316/files#r327393530

Reply:
Ah, I didn't go deep enough.  👍